docs/api.md

```diff
@@ -1206,6 +1232,19 @@ If `key` is an array, interpret all the elements as strings.
 `.string('_')` will result in non-hyphenated arguments being interpreted as strings,
 regardless of whether they resemble numbers.
 
+<a name="then"></a>.then()
```

Comment:
to make this as powerful as `.parse()` we should allow a context object to be passed in, this is useful for passing state around (as an example we use this for chat bots at npm).